### PR TITLE
Added the expire date to course page

### LINKED ISSUE #10: Show date that portfolios will be marked read-only
**[FR12]**
**Description:**
As an instructor, I want to see the date that a portfolio will be marked read-only so that I know when I have to get my evaluations done.

**Priority:** Medium
**Effort:** 1
**Acceptance Criteria:**
1. [x] Next to each active portfolio there should be a date listing when the portfolio will become read-only.

**Assigned:** Peter Bifone

### PR DESCRIPTION
**Pull request by PeterBifone**
Resolves #10

### CODE REVIEW

File: src/main/migrations/20191023190556_add_expire_and_read_only.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+
+exports.up = function(knex) {
+	return Promise.all([
+		knex.schema.table('portfolio', function(table) {
+		table.datetime('expireDate').notNull();
```
Comment (spigo900) [inline src/main/migrations/20191023190556_add_expire_and_read_only.js]: Fix indents? This and next two lines should be indented one level.

File: src/main/migrations/20191023190556_add_expire_and_read_only.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+		table.integer('readOnly').notNull();
+	})
+	]);
+};
+
+exports.down = function(knex) {
+  	return Promise.all([knex.schema.table('portfolio', function(table) {
+		table.datetime('expireDate')
+		table.dropColumn('readOnly')
+	})
```
Comment (spigo900) [inline src/main/migrations/20191023190556_add_expire_and_read_only.js]: This line (15) should be indented a level

File: src/main/routes/course.js [hunk 3]
```diff
@@ -107,9 +110,22 @@ const course_new_page = async (res, department = false) => {
 /* GET course home page */
 router.route('/')
 	.get(html.auth_wrapper(async (req, res, next) => {
+		const artifacts_active 			=  
+			{ arti : await Artifact.query().eager('[evaluations, owner.owner.[semester,owner.owner]]'),
```
Comment (spigo900) [inline src/main/routes/course.js]: No space after `arti` to be consistent with the rest of the codebase, and name should maybe be complete (`artifact`).

File: src/main/routes/course.js [hunk 4]
```diff
@@ -107,9 +110,22 @@ const course_new_page = async (res, department = false) => {
...
+			  formatDate:function(){
+			 	console.log(this)
+			 	return this.toLocaleString('default', {month: 'short', day: 'numeric', year : 'numeric' } ); 
+			  
+			  }
+			}
+		// console.log("HEY")
```
Comment (spigo900) [inline src/main/routes/course.js]: kill debug messages?

File: src/main/views/course/index.html [hunk 5]
```diff
@@ -8,29 +8,30 @@
...
 			<tr>
+				
 				<th>Course</th>
 				<th>Semester</th>
 				<th>Year</th>
 				<th>Artifact Progress <i class="material-icons">help</i></th>
 				<th>Due Date</th>
 				<th></th>
 			</tr>
+			{{#artifacts_active}}
```
Comment (spigo900) [inline src/main/views/course/index.html:18]: Indent to match next line?

File: src/main/views/course/index.html [hunk 6]
```diff
@@ -8,29 +8,30 @@
...
-				<td><a href="/course/123">edit <i class="material-icons">edit</i></a></td>
-			</tr>
-			<tr>
-				<td>CS499</td>
-				<td>Fall</td>
-				<td>2019</td>
-				<td>12 / 12</td>
-				<td>Dec 5, 2019</td>
-				<td><a href="/course/123">edit <i class="material-icons">edit</i></a></td>
+				<td><span class="uppercase">{{owner.owner.owner.owner.identifier}}</span>
```
Comment (spigo900) [inline src/main/views/course/index.html:21]: not an issue, but wow so many owners 💯

File: src/main/views/course/index.html [hunk 7]
```diff
@@ -8,29 +8,30 @@
...
+				<td><span class="capitalize">{{owner.owner.semester.value}}</span></td>
+				<td>{{owner.owner.year}}</td>
+				<td>{{evaluations.length}}</td>
+				{{#owner.owner.expireDate}}
+				<td>{{formatDate}}</td>
+				{{/owner.owner.expireDate}}
+				<td><a href="/course/{{id}}">edit <i class="material-icons">edit</i></a></td>
 			</tr>
+
+				{{/arti}}
```
Comment (spigo900) [inline src/main/views/course/index.html]: Just remember that if you rename `arti` (above) to a longer name then this should also be renamed to match.

File: src/test/cases/models/portfolio/portfolio.js [hunk 8]
```diff
@@ -16,7 +16,10 @@ describe('Model - CoursePortfolio', () => {
 				semester_term_id: 1,
 				num_students: 5,
 				section: 1,
-				year: 2019
+				year: 2019,
+				expireDate: new Date(Date.parse("2019-12-24T05:00:00.000Z")),
+				readOnly: 0,
```
Comment (spigo900) [inline src/test/cases/models/portfolio/portfolio.js]: Is `readOnly` a number? Shouldn't it be a boolean?

File: src/main/migrations/20191023190556_add_expire_and_read_only.js [hunk 9]
```diff
@@ -0,0 +1,17 @@
+
+exports.up = function(knex) {
+	return Promise.all([
+		knex.schema.table('portfolio', function(table) {
+		table.datetime('expireDate').notNull();
+		table.integer('readOnly').notNull();
```
Comment (spigo900) [inline src/main/migrations/20191023190556_add_expire_and_read_only.js]: FYI `table.boolean('readOnly')` is an option

File: src/main/views/course/index.html (hunk 6 above)
Comment (PeterBifone) [reply src/main/views/course/index.html:21]: yes

File: src/main/views/course/index.html (hunk 5 above)
Comment (PeterBifone) [reply src/main/views/course/index.html:18]: moved {{#artifacts}} (originally {{#arti}} ) to match level of indentation as {{#artifacts_active}}

File: src/main/migrations/20191023190556_add_expire_and_read_only.js [hunk 10]
```diff
@@ -0,0 +1,19 @@
+
+exports.up = function(knex) {
+	return Promise.all([
+		knex.schema.table('portfolio', function(table) {
+			table.datetime('expireDate')
+			table.boolean('readOnly')
+
```
Comment (spigo900) [inline src/main/migrations/20191023190556_add_expire_and_read_only.js]: Extra blank line?

File: src/main/routes/course.js [hunk 11]
```diff
@@ -107,9 +110,18 @@ const course_new_page = async (res, department = false) => {
 /* GET course home page */
 router.route('/')
 	.get(html.auth_wrapper(async (req, res, next) => {
+		const artifacts_active 			=  
+			{ artifact: await Artifact.query().eager('[evaluations, owner.owner.[semester,owner.owner]]'),
+			  formatDate:function(){
+			 	return this.toLocaleString('default', {month: 'short', day: 'numeric', year : 'numeric' } ); 
+
```
Comment (spigo900) [inline src/main/routes/course.js]: Another extra blank line?

File: src/main/routes/course.js [hunk 12]
```diff
@@ -107,9 +110,18 @@ const course_new_page = async (res, department = false) => {
 /* GET course home page */
 router.route('/')
 	.get(html.auth_wrapper(async (req, res, next) => {
+		const artifacts_active 			=  
+			{ artifact: await Artifact.query().eager('[evaluations, owner.owner.[semester,owner.owner]]'),
+			  formatDate:function(){
+			 	return this.toLocaleString('default', {month: 'short', day: 'numeric', year : 'numeric' } );
```
Comment (spigo900) [inline src/main/routes/course.js]: Probably want the end to look like `year: 'numeric'});` to match the beginning (there's no space after `(` or `(` and usually no space in `key: value` between `key` and `:`).

File: src/main/views/course/index.html [hunk 13]
```diff
@@ -8,29 +8,30 @@
 		<h2>Active Courses <i class="material-icons">help</i></h2>
 		<table>
 			<tr>
+
```
Comment (spigo900) [inline src/main/views/course/index.html]: Kill?

File: src/main/views/course/index.html [hunk 14]
```diff
@@ -8,29 +8,30 @@
...
+					{{owner.owner.owner.number}}</td>
+				<td><span class="capitalize">{{owner.owner.semester.value}}</span></td>
+				<td>{{owner.owner.year}}</td>
+				<td>{{evaluations.length}}</td>
+				{{#owner.owner.expireDate}}
+				<td>{{formatDate}}</td>
+				{{/owner.owner.expireDate}}
+				<td><a href="/course/{{id}}">edit <i class="material-icons">edit</i></a></td>
 			</tr>
+
```
Comment (spigo900) [inline src/main/views/course/index.html]: Kill extra? It doesn't match the beginning.

**Review by spigo900 — APPROVED** [review]
Looks good!

**Pull request merged**